dfc/keeper.go

```diff
@@ -208,11 +208,15 @@ func (r *proxykalive) keepalive(err error) (stopped bool) {
 		glog.Infof("proxy %s keepalive triggered by err %v", r.p.si.DaemonID, err)
 	}
 
+	smapLock.Lock()
+
 	if r.p.primary {
+		smapLock.Unlock()
 		return r.primarykeepalive()
 	}
 
 	if r.p.smap.ProxySI == nil || !r.timedOut(r.p.smap.ProxySI.DaemonID) {
+		smapLock.Unlock()
 		return
 	}
 
```

Comment:
Why smapLock is not Unlocked here?

Reply:
yup, that's why i think i need to run the whole tests, i only verified that all daemon starts
great eyes, that's why my code need reviews, thanks